packs/hpe-icsp/actions/lib/icsp.py

```diff
@@ -63,14 +63,31 @@ def get_sessionid(self):
 
         # added here due to the requirement of the session id
         self.base_headers = {'Auth': self.icsp_sessionid,
-                   'X-Api-Version': self.icsp_apiv}
+                             'X-Api-Version': self.icsp_apiv}
 
         return results["sessionID"]
 
     def extract_id(self, joburi):
         jobid = str(joburi)
         return int(jobid.split("/")[-1])
 
+    def get_mids(self, ids, idtype):
+        endpoint = "/rest/os-deployment-servers"
+        getresults = self.icsp_get(endpoint)
+        servers = getresults["members"]
+        mids = []
+        for id in ids:
+            for server in servers:
+                if idtype == 'serialnumber':
+                    if ((server["serialNumber"] == id) and
```

Comment:
This double loop and nesting is a bit hard to read and probably also not that efficient...

I think you could probably get rid of that outter loop and do something like:

```python
for server in servers:
    if (idtype == 'serialnumber' and server["serialNumber"] in ids) and (server["mid"] not in mids):
        mids.append(int(server["mid"]))
    elif (idtype == 'uuid' and server["uuid"] in ids) and (server["mid"] not in mids):
        mids.append(int(server["mid"]))
```